lxd/db/cluster/secrets.go

```diff
@@ -301,3 +304,97 @@ func deleteSecretsByID(ctx context.Context, tx *sql.Tx, ids ...int) error {
 
 	return nil
 }
+
+// GetAllBearerIdentitySigningKeys returns a map of identity ID to token signing keys.
+// It should only be used to refresh the identity cache.
+// Any identities with more than one signing key (this should never happen) are omitted from the returned map and added
+// to the list of identity IDs with multiple signing keys.
+// This so that the identity cache refresh handling can handle this at the call site (e.g. so as not to prevent
+// refreshing the cache for all identities).
+func GetAllBearerIdentitySigningKeys(ctx context.Context, tx *sql.Tx) (identityIDToSigningKey map[int]AuthSecretValue, identityIDsWithMultipleSigningKeys []int, err error) {
```

Comment:
I don't really like how we have special handling in multiple places (and call sites) for something that should never happen.

Rather than bake into the code detection and handling for a scenario that shouldn't even happen, why do we put a trigger in the DB to prevent it, that way if we ever try to do it accidentally it will fail at source.